### ISSUE TITLE
Setting language for Labels Documentation

### ISSUE THREAD

**Issue by gDev95** [mapbox/mapbox-gl-js-docs#41]
**mapbox-gl-js version: 0.53.1**:

### Question

I am trying to set the language of the labels according to some locale that may differ. Your base example and I think most prominent
```
<script>
mapboxgl.accessToken = 'pk.eyJ1IjoiY2FtcGFuZGFnbWJoIiwiYSI6ImNqc2J1eGVqaTBjaGc0NG15NDBqN2lseDIifQ.hw0232CbXUT5vc-gzRD1PQ';
var map = new mapboxgl.Map({
container: 'map',
style: 'mapbox://styles/mapbox/light-v9',
center: [16.05, 48],
zoom: 2.9
});
 
document.getElementById('buttons').addEventListener('click', function(event) {
var language = event.target.id.substr('button-'.length);
// Use setLayoutProperty to set the value of a layout property in a style layer.
// The three arguments are the id of the layer, the name of the layout property,
// and the new property value.
map.setLayoutProperty('country-label-lg', 'text-field', ['get', 'name_' + language]);
});
 
</script>
```
seems like a good start and works but little additional information can be found. For example I using on of mapbox's out-of-the-box map themes `mapbox://styles/mapbox/light-v9`. What is the id to change the state labels or city labels? 

And also what convention are you following here. After some more searching I find some Android Example and some iOS development example regarding this topic. 

```
Layer singleMapLayer = map.getLayer("place-city-lg-n");
singleMapLayer.setProperties(textField("{name_ru}"));
```

and 

```
let layer = style.layer(withIdentifier: "place-city-sm") as! MGLSymbolStyleLayer
let spanish = Locale(identifier: "es")
layer.text = layer.text.mgl_expressionLocalized(into: spanish)
```

I have yet to find an explanation why  its `place-city-lg-n` instead of `place-city-lg` 
Nor can I find anything about what is the difference between `place-city-sm` and `place-city-lg-n` . 
Furthermore `place-city-lg-n` vs. `country-label-lg`. The conventions are confusing, especially without a list of defined label ids for them.

Would be super awesome if this would be clarified. I feel for most usages with your awesome base themes it would be great to have a list of id's for the labels. So we can dynamically manipulate them. 

### Links to related documentation

[Language Switch](https://docs.mapbox.com/mapbox-gl-js/example/language-switch/)
[Change label language dynamically](https://docs.mapbox.com/help/troubleshooting/change-language/)

Hope you guys can help

**Comment by reustle** [comment]
I'd like to chime in here with some similar issues. This seems to be the most relevant demo () and yet it only changes the country label and nothing else.

<img width="523" alt="image" src="https://user-images.githubusercontent.com/304560/74794486-a7e60400-5306-11ea-854c-ad6f3fbc1ac5.png">

In that demo, they're setting the `country-label` property specifically.

```
map.setLayoutProperty('country-label', 'text-field', ['get','name_ja']);
```

If I traverse through `map.getStyle().layers` of my own map, and filter out only those of `type: 'symbol'` then I see these:

<img width="618" alt="image" src="https://user-images.githubusercontent.com/304560/74794613-09a66e00-5307-11ea-91e1-16f338cbc525.png">

I'm using the default `dark` theme from mapbox, so is it recommended to change each and every one of these? I haven't seemed to find any higher level 'lang' field that covers all labels.

Thank you!

**Comment by siakaramalegos** [comment]
@reustle's similar issue is the same as mine. My map needs to show both English and the local language, but no examples nor documentation of the list of labels exist. I may need to hack a similar solution, though it feels like this should be a feature - show a map in both your own language but also the language of a region.

**Comment by andrewharvey** [comment]
> @reustle's similar issue is the same as mine. My map needs to show both English and the local language, but no examples nor documentation of the list of labels exist. I may need to hack a similar solution, though it feels like this should be a feature - show a map in both your own language but also the language of a region.

Have you seen the example at https://docs.mapbox.com/mapbox-gl-js/example/display-and-style-rich-text-labels/? I think this could also be added as an option in https://github.com/mapbox/mapbox-gl-language/.

**Comment by siakaramalegos** [comment]
See the comment before mine to which I was referring. That only shows the country labels which isn't a very broad use case. Looping over all layers seems excessive.

**Comment by andrewharvey** [comment]
> See the comment before mine to which I was referring. That only shows the country labels which isn't a very broad use case. Looping over all layers seems excessive.

https://github.com/mapbox/mapbox-gl-language/ is the best option, it loops through layers for you, abstracting away the implementation detail.

The example the OP here refers to has the description

> Using setLayoutProperty to switch languages dynamically. For a more complete solution see the mapbox-gl-language plugin.

So I think the current example is good for showing in simple cases how to change a label language but mapbox-gl-language is best for anything serious.

**Comment by andrewharvey** [comment]
Thinking this over more, I think the example at https://docs.mapbox.com/mapbox-gl-js/example/language-switch/ should be renamed something like "change the language of a layer" as it shows a minimal example, however we should have a new example called "Change a map's language" which uses mapbox-gl-language which is a much better solution for an out of the box set the map language.

It seems like a lot of people are finding this example and getting frustrated it's not a complete solution and might not read the description, so better to have an example for a complete solution.

**Comment by reustle** [comment]
I agree that it would help. I still feel that it is odd that there isn't some native level method to call for language switching, since the tiles / layers support it. For comparison, switching language on Google Maps is _very_ straightforward https://developers.google.com/maps/documentation/javascript/localization

**Comment by andrewharvey** [comment]
> I agree that it would help. I still feel that it is odd that there isn't some native level method to call for language switching, since the tiles / layers support it. For comparison, switching language on Google Maps is very straightforward https://developers.google.com/maps/documentation/javascript/localization

There is, you can just do this once you've included the MapboxLanguage plugin

```js
map.addControl(new MapboxLanguage({
  defaultLanguage: 'ko'
}));
```

**Comment by siakaramalegos** [comment]
> Thinking this over more, I think the example at https://docs.mapbox.com/mapbox-gl-js/example/language-switch/ should be renamed something like "change the language of a layer" as it shows a minimal example, however we should have a new example called "Change a map's language" which uses mapbox-gl-language which is a much better solution for an out of the box set the map language.
> 
> It seems like a lot of people are finding this example and getting frustrated it's not a complete solution and might not read the description, so better to have an example for a complete solution.

This sounds right. Thanks for the full language change example here. I'd still like a way to show both, but I think I'll have to loop through the layers to do it.
